src/Form/DeliveryReferencedContentConfirmForm.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace Drupal\delivery\Form;
+
+use Drupal\Core\Form\ConfirmFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\delivery\DeliveryCartReferencedContent;
+use Drupal\delivery\DeliveryCartService;
+use Drupal\workspaces\Entity\Workspace;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class DeliveryReferencedContentConfirmForm extends ConfirmFormBase {
+
+  protected $deliveryCart;
+
+  protected $cart = [];
+
+  protected $cartEntities = [];
+
+  /**
+   * DeliveryEmptyCartConfirmForm constructor.
+   *
+   * @param DeliveryCartService $delivery_cart
+   */
+  public function __construct(DeliveryCartService $delivery_cart) {
+    $type = 'node';
```

Comment:
To have a generic implementation, let's don't check only for node items in the cart.